src/components/EditCoursePage/index.jsx

```diff
@@ -55,11 +56,40 @@ class EditCoursePage extends React.Component {
   }
 
   prepareSendCourseRunData(courseData) {
-    const { courseSubmitInfo: { targetRun } } = this.props;
+    const {
+      courseInfo: {
+        data: {
+          entitlements,
+        },
+      },
+      courseSubmitInfo: {
+        targetRun,
+      },
+    } = this.props;
+
     const sendCourseRuns = [];
-    // Don't send any courses in review - backend will reject them
-    const modifiedCourseRuns = courseData.course_runs.filter(run => (
-      !IN_REVIEW_STATUS.includes(run.status)
+    const initialCourseRunValues = this.buildCourseRuns();
+    const entitlement = entitlements && entitlements[0];
+    const initialMode = entitlement && entitlement.mode;
+    const initialPrice = entitlement && entitlement.price;
+
+    const modifiedCourseRuns = courseData.course_runs.filter((run, i) => (
+      // If we are submitting a run for review or re-publishing a run, it should
+      // always get through to the backend
+      (targetRun && (run.key === targetRun.key)) ||
+      // Don't send any courses in review - backend will reject them
+      (
+        !IN_REVIEW_STATUS.includes(run.status) &&
+        (
+          // Only send runs if they have changed OR the course mode or price has
+          // changed and the run is "archived"
```

Comment:
Maybe just for the dummies, say *why* we care about mode and price ("they are used on the backend to determine course run level info" or something).